### PR TITLE
fix(datastore): use '!' as LIKE escape char to fix MySQL 1064 error

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

`GetSpeciesLastDetectionDateBefore` in the v2only datastore built a `LIKE ... ESCAPE` clause with a backslash escape character. The Go string `"... ESCAPE '\\'"` renders to the SQL text `ESCAPE '\'`, and MySQL's default `sql_mode` treats a lone backslash in a string literal as an escape character, so the string literal never terminates and the server returns `Error 1064 (42000)`. SQLite does not treat backslash as special, so only MySQL backends were affected and the SQLite-only test path masked the bug. The failure breaks the new-species-tracker novelty / absence-return detection path on MySQL.

The fix switches the `LIKE` escape character to `!`, which is not special in any SQL dialect's string literals, and escapes `%`, `_`, and `!` in the queried scientific name accordingly. The generated SQL is now valid and identical on MySQL, SQLite, and Postgres. This is the only `ESCAPE '\\'` occurrence in the datastore.

## Changes

- `internal/datastore/v2only/datastore.go`: use `!` as the `LIKE` escape character and escape the pattern with it; document why backslash must not be used.
- Add a MySQL testcontainer integration test (`//go:build integration`) that reproduces the 1064 error against a real MySQL backend, plus a portable SQLite functional test covering exact match, the legacy `Scientific_Common` concatenated-label prefix match, literal `%`/`_` escaping, the strict cutoff boundary, and the false-positive review exclusion.
- Wire `./internal/datastore/v2only` into the CI `testcontainer-tests` job so the new integration test actually runs.

## Test Plan

- [x] `go test ./internal/datastore/v2only/` (SQLite) passes
- [x] `go test -tags=integration ./internal/datastore/v2only/` reproduces Error 1064 before the fix and passes after (verified against MySQL 8.0)
- [x] `golangci-lint run` clean (full project)
- [x] `internal/analysis/species` (the caller) passes under `-race`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved species name matching for MySQL by correcting LIKE escaping behavior, preventing SQL syntax errors and ensuring `%`, `_`, and legacy label formats are treated as literals.

* **Tests**
  * Added/expanded regression tests for `GetSpeciesLastDetectionDateBefore` on both SQLite and MySQL, including cutoff edge cases, legacy prefix matching, and special-character escaping.

* **Chores**
  * Updated CI integration test runs to include the additional datastore v2-only package.

### CODE REVIEW

File: internal/datastore/v2only/datastore.go [hunk 1]
```diff
@@ -2755,15 +2755,27 @@ func (ds *Datastore) GetSpeciesLastDetectionDateBefore(ctx context.Context, scie
...
 
-	escapedScientificName := strings.NewReplacer(`\`, `\\`, `%`, `\%`, `_`, `\_`).Replace(scientificName)
+	// Escape LIKE metacharacters using '!' as the escape character. '!' is not
+	// special in any SQL dialect's string literals, so the generated SQL is
+	// identical and valid on MySQL, SQLite, and Postgres. A backslash escape
+	// ('\') must NOT be used here: MySQL's default sql_mode treats a lone
+	// backslash in a string literal as an escape character, so "ESCAPE '\'"
+	// swallows the closing quote and raises a syntax error (Error 1064). SQLite
+	// does not treat backslash as special, which is why this only broke MySQL.
+	escapedScientificName := strings.NewReplacer(`!`, `!!`, `%`, `!%`, `_`, `!_`).Replace(scientificName)
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/datastore/v2only/datastore.go]: Good call, fixed in 0f2a1c56c. Hoisted it to a package-level var (scientificNameLikeEscaper) so the matcher is precomputed once; strings.Replacer is immutable and safe for concurrent use.
